crates/ty_python_semantic/src/semantic_index/place.rs

```diff
@@ -14,91 +15,303 @@ use crate::Db;
 use crate::ast_node_ref::AstNodeRef;
 use crate::node_key::NodeKey;
 use crate::semantic_index::visibility_constraints::ScopedVisibilityConstraintId;
-use crate::semantic_index::{SemanticIndex, SymbolMap, semantic_index};
+use crate::semantic_index::{PlaceSet, SemanticIndex, semantic_index};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, salsa::Update)]
+pub(crate) enum PlaceExprSubSegment {
+    /// A member access, e.g. `y` in `x.y`.
+    Member(ast::name::Name),
+    /// An integer index access, e.g. `1` in `x[1]`.
+    IntSubscript(ast::Int),
+    /// A string index access, e.g. `"foo"` in `x["foo"]`.
+    StringSubscript(String),
```

Comment:
This is maybe a bit pedantic, but would this be slightly clearer (if it's correct)?
```suggestion
    /// A member access, e.g. `.y` in `x.y`
    Member(ast::name::Name),
    /// An integer-based index access, e.g. `[1]` in `x[1]`
    IntSubscript(ast::Int),
    /// A string-based index access, e.g. `["foo"]` in `x["foo"]`
    StringSubscript(String),
```